journal-api/src/main/java/com/liferay/journal/util/JournalStringUtil.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.journal.util;
+
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Pavel Savinov
+ */
+public class JournalStringUtil {
+
+	public static String shortenWithHtml(String string, int length) {
+		String shortString = StringUtil.shorten(string, length, "");
```

Comment:
@pavel-savinov this logic does not make sense, in case string.length() <= length is true, we should just return. We should not try to change anything here, even the string contains unclosed tags, that is what the user inputted, we should not mess with the content. We should only fix the string with closing tags in case we actually shorten it for real.